### DISCUSSION TITLE
Two API requests are executed to display one page

### DISCUSSION THREAD

**Discussion by mirumirumi** [nuxt/nuxt#16590] [Questions]
In Nuxt 3, code written within the `<script>` tag of each component will be executed twice, once on the server side and once on the client side (some, such as `onMounted()`, are not).
I understand that this is necessary to control the real DOM on the client side while using the rendered HTML on the server side.

However, I am not convinced that data fetching should be done twice to display a page once.
It does not seem reasonable to go to the trouble of pre-creating HTML in SSG, only to have the data fetched again each time a user visits the page (If it were SSR instead of SSG, the situation would be the same, I think).

For example, I tried to use `if (process.server)` for this problem, but instead of getting a hydration error, nothing happens simply because the necessary data is not passed to the template in the first place.

When using serverless functions such as AWS Lambda, we also run into the cold start issue as we need to return a response each time a user views a page.

Does this behavior disturb many of the problems that SSG is supposed to solve?
Or if I am wrong, please let me know.

**Comment by hamidooChe** [comment]
Fetching data with useFetch(url, { server: false }) will disable the first call on serverside

**Reply by mirumirumi** [reply]
Does that mean that `npm run generate` will produce an empty page?
